src/appleseed/renderer/modeling/aov/depthaov.cpp

```diff
@@ -58,34 +64,53 @@ namespace
     //
 
     class DepthAOVAccumulator
-      : public AOVAccumulator
+      : public UnfilteredAOVAccumulator
     {
       public:
-        explicit DepthAOVAccumulator(const size_t index)
-          : AOVAccumulator(index)
+        explicit DepthAOVAccumulator(Image& image)
+          : UnfilteredAOVAccumulator(image)
         {
         }
 
-        virtual void reset() override
+        virtual void on_tile_begin(
+            const Frame&                frame,
+            const size_t                tile_x,
+            const size_t                tile_y) override
         {
-            m_depth = numeric_limits<float>::max();
+            UnfilteredAOVAccumulator::on_tile_begin(frame, tile_x, tile_y);
+
+            get_tile().clear(Vector2f(std::numeric_limits<float>::max()));
         }
 
         virtual void write(
-            const ShadingPoint&     shading_point,
-            const Camera&           camera) override
+            const PixelContext&         pixel_context,
+            const ShadingPoint&         shading_point,
+            const ShadingComponents&    shading_components,
+            ShadingResult&              shading_result) override
         {
-            if (shading_point.hit())
-                m_depth = static_cast<float>(shading_point.get_distance());
-        }
+            const Vector2i& pi = pixel_context.get_pixel_coords();
 
-        virtual void flush(ShadingResult& result) override
-        {
-            result.m_aovs[m_index] = Color4f(m_depth, m_depth, m_depth, 1.0f);
-        }
+            // Ignore samples outside the tile.
+            if (outside_tile(pi))
+                return;
 
-      private:
-        float m_depth;
+            float* p = reinterpret_cast<float*>(
+                get_tile().pixel(pi.x - m_tile_origin_x, pi.y - m_tile_origin_y));
+
+            const float min_sample_squared_distance = p[1];
+            const float sample_squared_distance =
+                squared_distace_to_pixel_center(pixel_context.get_sample_position());
```

Comment:
Typo: *distace* -> *distance*

Also, for the sake of consistency, may I suggest *square* instead of *squared*?
 I.e. `square_distance_to_pixel_center()`